File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/JavaScript/_019)_Maximum_Difference.js
function maxDiff(arr) {
    let n = arr.length;
    let res = arr[1] - arr[0];
    for (let i = 0; i < n; i++)
        for (let j = i + 1; j < n; j++) res = Math.max(res, arr[j] - arr[i]);
    return res;
}

let arr = [2, 3, 10, 6, 4, 8, 1];
console.log(`Maximum Difference = ${maxDiff(arr)}`);

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_019)_Maximum_Difference.dart
int maxDiff(arr) {
  var n = arr.length;
  var res = arr[1] - arr[0];

  for (int i = 0; i < n; i++) {
    for (int j = i + 1; j < n; j++) {
      if (arr[j] - arr[i] > res) {
        res = arr[j] - arr[i];
      }
    }
  }
  return res;
}

void main() {
  var arr = [2, 3, 10, 6, 4, 8, 1];
  var diff = maxDiff(arr);
  print('Leaders in the array are: $diff');
}
